### PR TITLE
Add tests for holyday type and omer day and some small code refactoring

### PR DESCRIPTION
**Pull request by tsvi**

### CODE REVIEW

File: hdate/hdate_julian.py [hunk 1]
```diff
@@ -36,10 +36,13 @@ def _days_from_3744(hebrew_year):
     week_day = parts_left_in_week / PARTS_IN_DAY
 
     # Special cases of Molad Zaken
-    if ((leap_left < 12 and week_day == 3 and
-         parts_left_in_day >= get_chalakim(9 + 6, 204)) or
-            (leap_left < 7 and week_day == 2 and
-             parts_left_in_day >= get_chalakim(15 + 6, 589))):
+    if (leap_left < 12 and week_day == 3 and
```
Comment (royi1000) [inline hdate/hdate_julian.py]: this is a dangerous change,
if somehow we will change the if order we will have a bug,
I think it's better in the old code

File: hdate/htables.py [hunk 2]
```diff
@@ -25,7 +25,7 @@
         0, 0, 0, 0, 11, 0, 0, 0, 0, 0,
         0, 0, 0, 0, 0, 0, 0, 0, 0, 33],
     [  # Adar
-        0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 34, 0, 0, 0,
```
Comment (royi1000) [inline hdate/htables.py]: you didn't added the day in the table  (״יום פטירת משה רבינו״)

Comment (tsvi) [inline hdate/htables.py]: It is already in the table: 
יום הזיכרון לחללי מערכות ישראל שמקום קבורתם לא נודע

File: hdate/hdate_julian.py (hunk 1 above)
Comment (tsvi) [inline hdate/hdate_julian.py]: Ok, reverting and adding an exception in the linter

File: hdate/htables.py (hunk 2 above)
Comment (royir) [inline hdate/htables.py]: OK

**Comment by tsvi** [comment]
Added commits as requested

**Pull request merged**